velox/exec/tests/SortBufferTest.cpp

```diff
@@ -485,34 +470,152 @@ DEBUG_ONLY_TEST_F(SortBufferTest, reserveMemoryGetOutput) {
       &spillConfig,
       &spillStats);
 
+  const int numInputs = 10;
+  const int numSpilledInputs = 10 / 2;
+  std::atomic_int processedInputs{0};
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::SortBuffer::addInput",
+      std::function<void(SortBuffer*)>(([&](SortBuffer* sortBuffer) {
+        if (processedInputs++ != numSpilledInputs) {
+          return;
+        }
+        ASSERT_GT(sortBuffer->pool()->usedBytes(), 0);
+        sortBuffer->spill();
+        ASSERT_EQ(sortBuffer->pool()->usedBytes(), 0);
+      })));
+
   const std::shared_ptr<memory::MemoryPool> fuzzerPool =
-      memory::memoryManager()->addLeafPool("spillSource");
+      memory::memoryManager()->addLeafPool("spillDuringInput");
   VectorFuzzer fuzzer({.vectorSize = 1024}, fuzzerPool.get());
+  uint64_t totalNumInput{0};
+
+  ASSERT_EQ(memory::spillMemoryPool()->stats().usedBytes, 0);
+  const auto peakSpillMemoryUsage =
```

Comment:
shouldnt this be zero at this point too? since the pool hasent been used for anything yet

Reply:
The test might run with others in the same process.